src/main/java/org/cyclonedx/gradle/SbomBuilder.java

```diff
@@ -151,6 +153,25 @@ private Metadata buildMetadata(final SbomComponent parentComponent) {
         return metadata;
     }
 
+    private void addBuildSystemMetaData(Component component) {
+        if (task.getIncludeBuildSystem().get()) {
+            String url;
+            if (task.getBuildSystemEnvironmentVariable().isPresent()
+                    && task.getBuildSystemEnvironmentVariable().get() != null) {
+                url = EnvironmentUtils.getBuildURI(
+                        task.getBuildSystemEnvironmentVariable().get());
+            } else {
+                url = EnvironmentUtils.getBuildURI();
+            }
+            if (url != null) {
```

Comment:
The addExternalReference method would not add the external reference if the url is not a valid Url. I think it makes no difference if it is null, or just an invalid url. 

In case I am wrong, then we should also check if the url is blank imo

Reply:
Quick glance through the code - I think if the url is blank it will likely still add it to the model.